crates/pop-parachains/src/build.rs

```diff
@@ -1,13 +1,498 @@
+/// * `binary_path` - The path to the node binary executable that contains the `build-spec` command.
+/// * `chain_spec_file_name` - The name of the chain specification file to be generated.
+pub fn generate_raw_chain_spec(
+	path: Option<&Path>,
+	plain_chain_spec: &Path,
+	binary_path: &Path,
+	chain_spec_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !plain_chain_spec.exists() {
+		return Err(Error::MissingChainSpec(plain_chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "build-spec")?;
+	let raw_chain_spec = path.unwrap_or(Path::new("./")).join(chain_spec_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"build-spec",
+			"--chain",
+			&plain_chain_spec.display().to_string(),
+			"--disable-default-bootnode",
+			"--raw",
+		],
+	)
+	.stdout_path(raw_chain_spec.clone())
+	.run()?;
+	Ok(raw_chain_spec)
+}
+
+/// Export the WebAssembly runtime for the parachain.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-wasm` command.
+/// * `wasm_file_name` - The name of the wasm runtime file to be generated.
+pub fn export_wasm_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	wasm_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-wasm")?;
+	let wasm_file = path.unwrap_or(Path::new("./")).join(wasm_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-wasm",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&wasm_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(wasm_file)
+}
 
+/// Generate the parachain genesis state.
+///
+/// # Arguments
+/// * `path` - Location of the parachain project.
+/// * `chain_spec` - Location of the raw chain specification file.
+/// * `binary_path` - The path to the node binary executable that contains the `export-genesis-state` command.
+/// * `genesis_file_name` - The name of the genesis state file to be generated.
+pub fn generate_genesis_state_file(
+	path: Option<&Path>,
+	chain_spec: &Path,
+	binary_path: &Path,
+	genesis_file_name: &str,
+) -> Result<PathBuf, Error> {
+	if !chain_spec.exists() {
+		return Err(Error::MissingChainSpec(chain_spec.display().to_string()));
+	}
+	check_command_exists(&binary_path, "export-genesis-state")?;
+	let genesis_file = path.unwrap_or(Path::new("./")).join(genesis_file_name);
+	cmd(
+		binary_path,
+		vec![
+			"export-genesis-state",
+			"--chain",
+			&chain_spec.display().to_string(),
+			&genesis_file.display().to_string(),
+		],
+	)
+	.run()?;
+	Ok(genesis_file)
+}
+
+/// Parses the node name from the Cargo.toml file located in the project path.
+fn parse_node_name(node_path: &Path) -> Result<String, Error> {
```

Comment:
This isnt specific to a node, its reading the package name from a cargo.toml file. This can be moved to `common` as `parse_package_name`.